### PR TITLE
aws opensearch: add 4xx and 5xx detectors (+ PR 413)

### PR DESCRIPTION
**Pull request by tchernomax**
you can check on: CMP-QYZ68Z PRJ-OUR2QS
between: 04/29/2022 01:00:00 am to 05/06/2022 01:00:00 pm

### CODE REVIEW

**Comment by tchernomax** [comment]
I just also added "ShardCount", to avoid:
```
this action would add [X] total shards, but this cluster currently has [XXX]/[YYY] maximum shards open
```

File: modules/integration_aws-elasticsearch/conf/02-4xx.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudwatchmetrics.html"
+value_unit: "%"
+
+signals:
+  A:
+    metric: 4xx
+  B:
+    metric: 2xx
+  signal:
+    # "OpenSearchRequests" can be inferiore to 4xx (I don't know how), it's not the sum of 2xx, 3xx, 4xx and 5xx… so I juste compute
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/02-4xx.yaml:17]: I've posted my understanding of this behavior here: https://stackoverflow.com/a/72818868/316805

TL;DR: `4xx` cannot be compared to `OpenSearchRequests`, but `5xx` can.

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudwatchmetrics.html"
+value_unit: "%"
+
+signals:
+  A:
+    metric: 5xx
+  B:
+    metric: 2xx
+  signal:
+    # "OpenSearchRequests" can be inferiore to 4xx (I don't know how), it's not the sum of 2xx, 3xx, 4xx and 5xx… so I juste compute
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:17]: I've posted my understanding of this behavior here: https://stackoverflow.com/a/72818868/316805

TL;DR: `4xx` cannot be compared to `OpenSearchRequests`, but `5xx` can.

File: modules/integration_aws-elasticsearch/conf/02-4xx.yaml (hunk 1 above)
Comment (pdecat) [reply modules/integration_aws-elasticsearch/conf/02-4xx.yaml:17]: Finally, I believe it is not worth monitoring `4xx` requests as they include unauthorized accesses ending up in `403` HTTP status codes.

File: modules/integration_aws-elasticsearch/detectors-elasticsearch.tf [hunk 3]
```diff
@@ -80,8 +80,10 @@ resource "signalfx_detector" "free_space" {
...
-    signal = free.scale(0.001).publish('signal')
+    data_node_free_storage   = data('FreeStorageSpace',       filter=filter('namespace', 'AWS/ES') and filter('stat', 'lower') and filter('NodeId', '*') and ${module.filtering.signalflow})${var.free_space_aggregation_function}${var.free_space_transformation_function}.scale(0.001)
+    warm_node_free_storage   = data('WarmFreeStorageSpace',   filter=filter('namespace', 'AWS/ES') and filter('stat', 'lower') and filter('NodeId', '*') and ${module.filtering.signalflow})${var.free_space_aggregation_function}${var.free_space_transformation_function}.scale(0.001)
+    master_node_free_storage = data('MasterFreeStorageSpace', filter=filter('namespace', 'AWS/ES') and filter('stat', 'lower') and filter('NodeId', '*') and ${module.filtering.signalflow})${var.free_space_aggregation_function}${var.free_space_transformation_function}.scale(0.001)
+    signal = union(data_node_free_storage, warm_node_free_storage, master_node_free_storage).publish('signal')
```
[the remark this answers is not in the document]
Comment (tchernomax) [reply modules/integration_aws-elasticsearch/detectors-elasticsearch.tf]: removed

File: modules/integration_aws-elasticsearch/conf/02-4xx.yaml (hunk 1 above)
Comment (tchernomax) [reply modules/integration_aws-elasticsearch/conf/02-4xx.yaml]: disabled + note in the readme

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml (hunk 2 above)
Comment (tchernomax) [reply modules/integration_aws-elasticsearch/conf/03-5xx.yaml]: fixed

**Review by pdecat — APPROVED** [review]
LGTM!
